internal/pkg/acl/acl_windows.go

```diff
@@ -0,0 +1,284 @@
+	//   If the function succeeds, the return value is ERROR_SUCCESS.
+	//   If the function fails, the return value is a nonzero error code defined in WinError.h.
+	//   Doesn't set last error.
+	procGetNamedSecurityInfoW = advapi32.MustFindProc("GetNamedSecurityInfoW")
+	// procSetNamedSecurityInfoW (https://learn.microsoft.com/en-us/windows/win32/api/aclapi/nf-aclapi-setnamedsecurityinfow)
+	//   If the function succeeds, the return value is ERROR_SUCCESS.
+	//   If the function fails, the return value is a nonzero error code defined in WinError.h.
+	//   Doesn't set last error.
+	procSetNamedSecurityInfoW = advapi32.MustFindProc("SetNamedSecurityInfoW")
+	// procSetEntriesInAclW (https://learn.microsoft.com/en-us/windows/win32/api/aclapi/nf-aclapi-setentriesinaclw)
+	//   If the function succeeds, the return value is ERROR_SUCCESS.
+	//   If the function fails, the return value is a nonzero error code defined in WinError.h.
+	//   Doesn't set last error.
+	procSetEntriesInAclW = advapi32.MustFindProc("SetEntriesInAclW")
+)
+
+// Trustee defines the user account, group account, or logon session to which an access control entry (ACE) applies.
+// https://msdn.microsoft.com/en-us/library/windows/desktop/aa379636.aspx
+type Trustee struct {
+	MultipleTrustee          *Trustee
+	MultipleTrusteeOperation int32
+	TrusteeForm              int32
+	TrusteeType              int32
+	Name                     *uint16
+}
+
+// ExplicitAccess defines access control information for a specified Trustee.
+// https://msdn.microsoft.com/en-us/library/windows/desktop/aa446627.aspx
+type ExplicitAccess struct {
+	AccessPermissions uint32
+	AccessMode        int32
+	Inheritance       uint32
+	Trustee           Trustee
+}
+
+// GrantSid creates an ExplicitAccess instance granting permissions to the provided SID.
+func GrantSid(accessPermissions uint32, sid *windows.SID) ExplicitAccess {
+	return ExplicitAccess{
+		AccessPermissions: accessPermissions,
+		AccessMode:        GRANT_ACCESS,
+		Inheritance:       SUB_CONTAINERS_AND_OBJECTS_INHERIT,
+		Trustee: Trustee{
+			TrusteeForm: TRUSTEE_IS_SID,
+			Name:        (*uint16)(unsafe.Pointer(sid)),
+		},
+	}
+}
+
+// Chmod changes the permissions of the specified file. Only the nine
+// least-significant bytes are used, allowing access by the file's owner, the
+// file's group, and everyone else to be explicitly controlled.
+func Chmod(name string, fileMode os.FileMode) error {
+	// https://support.microsoft.com/en-us/help/243330/well-known-security-identifiers-in-windows-operating-systems
+	creatorOwnerSID, err := windows.StringToSid("S-1-3-0")
+	if err != nil {
+		return err
+	}
+	creatorGroupSID, err := windows.StringToSid("S-1-3-1")
+	if err != nil {
+		return err
+	}
+	everyoneSID, err := windows.StringToSid("S-1-1-0")
+	if err != nil {
+		return err
+	}
+
+	mode := uint32(fileMode)
+	return Apply(
+		name,
+		true,
+		false,
+		GrantSid(((mode&0700)<<23)|((mode&0200)<<9), creatorOwnerSID),
```

Comment:
please comment these shifts for context

Reply:
although I "inherited" these from the external library that now lives inside agent code, I just added comments 🙂 feel free to un-resolve if you spot anything 🙂